instaloader/instaloadercontext.py

```diff
@@ -379,7 +379,19 @@ def get_json(self, path: str, params: Dict[str, Any], host: str = 'www.instagram
                 match = re.search(r'window\._sharedData = (.*);</script>', resp.text)
                 if match is None:
                     raise ConnectionException("Could not find \"window._sharedData\" in html response.")
-                return json.loads(match.group(1))
+                resp_json = json.loads(match.group(1))
+                entry_data = resp_json.get('entry_data')
+                post_or_profile_page = list(entry_data.values())[0] if entry_data is not None else None
+                if post_or_profile_page is None:
+                    raise ConnectionException("\"window._sharedData\" does not contain required keys.")
+                # If GraphQL data is missing in `window._sharedData`, search for it in `__additionalDataLoaded`.
+                if 'graphql' not in post_or_profile_page[0]:
+                    match = re.search(r'window\.__additionalDataLoaded\([^{]+{"graphql":({.*})}\);</script>',
+                                      resp.text)
+                    if match is None:
+                        raise ConnectionException("Could not find \"graphql\" in \"window.__additionalDataLoaded\"")
+                    post_or_profile_page[0]['graphql'] = json.loads(match.group(1))
```

Comment:
We tested this code and again encountered some mistakes. There are cases where `get_json()` is called to fetch html data without expecting any GraphQL data to be in the response. For example `Profile.from_id()` is not able to get the `root_rhx_gis` and Instaloader's logic for detecting username changes fails. In order to avoid these issues, `get_json()` must not raise an exception if no GraphQL data is found in the html query:

```python
                    if match is not None:
                        post_or_profile_page[0]['graphql'] = json.loads(match.group(1))
```

Reply:
done